src/get.rs

```diff
@@ -0,0 +1,72 @@
+use crate::config::get_config_toml;
+use crate::error::PasswordStoreError;
+use clap::Values;
+use keyring::Keyring;
+use nitrokey::GetPasswordSafe;
+use nitrokey::{connect, SLOT_COUNT};
+use rpassword::prompt_password_stdout;
+use std::io::Write;
+use std::process::{Command, Stdio};
+use std::thread::sleep;
+use std::time::Duration;
+
+fn get_default_username(service: &str) -> Result<String, PasswordStoreError> {
+    let toml = get_config_toml()?;
+    match toml.get(service) {
+        Some(username) => Ok(username.as_str().unwrap().into()),
+        None => Err(PasswordStoreError::NoDefaultUser(
+            format!("No default set for {}", service).into(),
+        )),
+    }
+}
+
+fn xclip(payload: &str) -> Result<(), PasswordStoreError> {
+    let xclip = Command::new("xclip")
+        .arg("-sel")
+        .arg("clip")
+        .stdin(Stdio::piped())
+        .spawn()?;
+    xclip.stdin.unwrap().write_all(payload.as_bytes())?;
+    Ok(())
+}
+
+pub fn get_password_from_nitrokey<'a>(service: &'a str, username: &'a str) -> Option<String> {
+    let device = match connect() {
+        Ok(device) => device,
+        _ => {
+            return None;
+        }
+    };
+    let pin = prompt_password_stdout("Nitro-key user pin: ").unwrap();
+    let password_safe = device.get_password_safe(&pin).unwrap();
+    for slot in 0..SLOT_COUNT {
+        if password_safe.get_slot_name(slot).unwrap() == service
+            && password_safe.get_slot_login(slot).unwrap() == username
+        {
+            return Some(password_safe.get_slot_password(slot).unwrap());
+        }
+    }
+    None
+}
+
+pub fn handle_get(mut values: Values) -> Result<(), PasswordStoreError> {
+    let service = values.next().unwrap();
+    let username = match values.next() {
+        Some(username) => username.into(),
+        None => get_default_username(service)?,
+    };
+    // search nitro-key for password
+    let password = match get_password_from_nitrokey(service, &username) {
+        Some(pw) => pw,
+        None => {
```

Comment:
It might be worth printing a message to indicate where the password was retrieved from.